OverlayPlugin.Core/Integration/FFXIVRepository.cs

```diff
@@ -274,6 +275,19 @@ public string GetLocaleString()
             }
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public bool WriteLogLineImpl(string line)
+        {
+            var plugin = GetPluginData();
+            if (plugin == null) return false;
+            var iocContainer = plugin.GetType().GetField("_iocContainer", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(plugin);
+            var getServiceMethod = iocContainer.GetType().GetMethod("GetService");
+            var logOutput = getServiceMethod.Invoke(iocContainer, new object[] { typeof(FFXIV_ACT_Plugin.Logfile.ILogOutput) }) as FFXIV_ACT_Plugin.Logfile.ILogOutput;

```

Comment:
This is maybe premature optimization, but maybe it's worth caching the `ILogOutput` in the `FFXIVRepository` here.

Reply:
The code itself is just proof of concept. There's a lot more null checking and error handling that needs to be done in general before this is actually ready for any sort of merge.